python/ql/src/Functions/ModificationOfParameterWithDefault.ql

```diff
@@ -93,7 +93,47 @@ class Mutation extends TaintSink {
     }
 }
 
-from TaintedPathSource src, TaintedPathSink sink
-where src.flowsTo(sink)
+/** copying prevents modification of the default value */
+class Copying extends Sanitizer {
+    Copying() { this = "Copy sanitizer"}
+
+    override predicate sanitizingDefinition(TaintKind kind, EssaDefinition def) {
+        (kind instanceof EmptyMutableValue
+        or
+        kind instanceof NonEmptyMutableValue) and
+        isCopied(def.(AssignmentDefinition).getValue())
+    }
+
+    predicate isCopied(ControlFlowNode value) {
+        value = Value::named("copy.copy").getACall()
+        or
+        value = Value::named("copy.deepcopy").getACall()
+        or
+        value.(AttrNode).getName() = "copy"
```

Comment:
I am not sure this is needed when we have the case below.

Reply:
Sounds to me like adding tests might supply the necessary certainty. :) 

As far as I can tell, this disjunct covers everything of the form `foo.copy`, whereas the next one covers `foo.copy()`. As such, I'm not sure this disjunct is useful.

Reply:
Well, it can take an awful lot of tests to distinguish equivalent things. I thought it better to get confirmation on the semantics :)

I also considered if the first form was in fact more general and therefor better (since it also seems to match the `foo.copy` part of `foo.copy()`), but having not read a bit about Python attributes, that would be a strange way to implement a copy functionality (so now you will tell me that of course someone did just that).